pkg/util/validation/limits.go

```diff
@@ -317,6 +318,7 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 		l.RulerMaxRuleGroupsPerTenantByNamespace = NewLimitsMap[int](nil)
 	}
 	f.Var(&l.RulerMaxRuleGroupsPerTenantByNamespace, "ruler.max-rule-groups-per-tenant-by-namespace", "Maximum number of rule groups per tenant by namespace. Value is a map, where each key is the namespace and value is the number of rule groups allowed in the namespace (int). On the command line, this map is given in a JSON format. The number of rule groups specified has the same meaning as -ruler.max-rule-groups-per-tenant, but only applies for the specific namespace. If specified, it supersedes -ruler.max-rule-groups-per-tenant.")
+	f.Var(&l.RulerProtectedNamespaces, "ruler.protected-namespaces", "List of namespaces that are protected from modification unless a special HTTP header is used. If a namespace is protected, it cannot be modified via the ruler's config API only read. The value is a list of strings, where each string is a namespace name. On the command line, this list is given as a comma-separated list.")
```

Comment:
This is actually a good shout - I'll create an issue and mark it as a good first issue to see if we can get it as an OSS contribution.

That being said... This is now the second time I've gotten confused about how `flagext.StringSlice` works (the last one was a couple of years ago so I quickly remembered when testing this), the flag must be repeated to register the multiple records is it not a comma-separated list of values.

Do you have a preference on what should I use? It seems like `flagext.StringSlice` was made a long time ago (cortex times) and is only used by 1 flag. Should I make a custom flag that allows us to do comma-separated values?

Reply:
I prefer comma-separated CLI flags. Setting mutiple entries of the same CLI flag in jsonnet is a pain (ever tried?).

Reply:
I can't recall (but I probably faced the same problem 😆) , but if there's anything to judge by, I can't even do that in an integration test because you can't repeat keys.

I'll create one.

Reply:
Turns out, we already had a flag for that 😄  Done in https://github.com/grafana/mimir/pull/8444/commits/d350a9388725128fb55037d3ff3a37beb93a0ce6

Reply:
I have also added your suggestion as an action item for: https://github.com/grafana/mimir-squad/issues/2136